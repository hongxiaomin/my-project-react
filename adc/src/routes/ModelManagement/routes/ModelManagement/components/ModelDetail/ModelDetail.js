import React from 'react'
import { connect } from 'react-redux'
import { injectIntl } from 'react-intl'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import Property from './PropertyDetail'
import classes from './ModelDetail.scss'
import {
  PAGE_STATUS,
} from '../../modules'

type Props = {
  dispatch: Function,
  rowDetail: Object,
  modelDetail: Object,
  trainingPercent: String,
  testingPercent: String,
  modelId: String,
  intl: Object,
};

export class TextField {
  constructor(key, value) {
    this.key = key
    this.value = value
  }

  getRenderNoWrap(btn) {
    return (
      <div className={classes.textField}>
        <div className={classes.key}>
          {this.key}
        </div>
        <div className={classes.valueNoWrap}>
          {this.value}
          {btn}
        </div>
      </div>
    )
  }
  getRender(btn) {
    return (
      <div className={classes.textField}>
        <div className={classes.key}>
          {this.key}
        </div>
        <div className={classes.value}>
          {this.value}
          {btn}
        </div>
      </div>
    )
  }
}

export class ModelDetail extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
    }
    this.toggelTrain = this.toggelTrain.bind(this)
    this.toggleTest = this.toggleTest.bind(this)
  }

  toggelTrain() {
    this.props.dispatch({
      type: PAGE_STATUS,
      payload: 'startTrainPage',
    })
  }
  toggleTest() {
    this.props.dispatch({
      type: PAGE_STATUS,
      payload: 'startTestPage',
    })
  }
  props: Props
  render() {
    const { intl } = this.props
    let traingingPercent
    let testingPercent
    if (this.props.trainingPercent !== '0' && this.props.modelId === this.props.rowDetail.id.toString()) {
      traingingPercent = `${(this.props.trainingPercent * 100).toFixed(0)}%`
    } else {
      traingingPercent = `${(this.props.modelDetail.trainingInfo.trainingProgress * 100).toFixed(0)}%`
    }
    if (this.props.testingPercent !== '0' && this.props.modelId === this.props.rowDetail.id.toString()) {
      testingPercent = `${(this.props.testingPercent * 100).toFixed(0)}%`
    } else {
      testingPercent = `${(this.props.modelDetail.testingInfo.testingProgress * 100).toFixed(0)}%`
    }
    return (
      <div className={classes.container}>
        <div className={classes['top-left']}>
          <span>
            <WrapperFormattedMessage id="modelManagement.ModelDetail.modelDetailInformation" />
          </span>
          <div>{this.props.rowDetail.name} / <WrapperFormattedMessage id="training" />
            {
              traingingPercent === '100%' ?
                <span style={{ color: 'rgb(102, 204, 0)' }}> <WrapperFormattedMessage id="complete" /> </span> :
                <span style={{ color: 'rgb(102, 204, 0)' }}> <WrapperFormattedMessage id="incomplete" /> </span>
            }
            / <WrapperFormattedMessage id="testing" />
            {
              testingPercent === '100%' ?
                <span style={{ color: 'rgb(102, 204, 0)' }}> <WrapperFormattedMessage id="complete" /> </span> :
                <span style={{ color: 'rgb(102, 204, 0)' }}> <WrapperFormattedMessage id="incomplete" /> </span>
            }
          </div>
        </div>
        <div className={classes.divContainer}>
          <div className={classes.singleDiv}>
            <div className={classes.title}>
              <WrapperFormattedMessage id="attribute" />
            </div>
            <div className={classes.detail}>
              {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.createTime' })}`, this.props.modelDetail.modelAttributes.createTime).getRenderNoWrap()}
              {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.pairedProcessID' })}`, this.props.modelDetail.modelAttributes.process.join(' / ')).getRenderNoWrap()}
              {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.pairedProductID' })}`, this.props.modelDetail.modelAttributes.product.join(' / ')).getRenderNoWrap()}
              {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.pairedDefect' })}`, this.props.modelDetail.modelAttributes.defect.join(' / ')).getRenderNoWrap()}
            </div>
            <Property
              data={this.props.modelDetail.modelAttributes}
              isDetail={this.props.rowDetail}
            />
          </div>
          <div className={classes.singleDiv}>
            <div className={classes.title}>
              <WrapperFormattedMessage id="training" />
            </div>
            {
              this.props.modelDetail.trainingInfo.trainingProgress === 1 ?
                <div className={classes.detail}>
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.trainingAt' })}`, this.props.modelDetail.trainingInfo.trainingBeginTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.doneAt' })}`, this.props.modelDetail.trainingInfo.trainingEndTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.progress' })}`, traingingPercent).getRender()}
                </div> :
                <div className={classes.detail}>
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.trainingAt' })}`, this.props.modelDetail.trainingInfo.trainingBeginTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.estimatedRemaining' })}`, this.props.modelDetail.trainingInfo.trainingEndTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.progress' })}`, traingingPercent).getRender()}
                </div>
            }
            <button
              style={{ width: '80px', position: 'absolute', bottom: '6%', right: '4%' }}
              type='button'
              onClick={this.toggelTrain}
              className={'pt-button pt-intent-primary'}
            >
              <WrapperFormattedMessage id="enter" />
            </button>
          </div>
          <div className={classes.singleDiv} style={{ borderRight: 'none' }}>
            <div className={classes.title}>
              <WrapperFormattedMessage id="testing" />
            </div>
            {
              this.props.modelDetail.testingInfo.testingProgress === 1 ?
                <div className={classes.detail}>
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.testingAt' })}`, this.props.modelDetail.testingInfo.testingBeginTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.doneAt' })}`, this.props.modelDetail.testingInfo.testingEndTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.progress' })}`, testingPercent).getRender()}
                </div> :
                <div className={classes.detail}>
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.testingAt' })}`, this.props.modelDetail.testingInfo.testingBeginTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.estimatedRemaining' })}`, this.props.modelDetail.testingInfo.TestingEndTime).getRender()}
                  {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.progress' })}`, testingPercent).getRender()}
                </div>
            }
            <button
              style={{ width: '80px', position: 'absolute', bottom: '6%', right: '4%' }}
              type='button'
              onClick={this.toggleTest}
              disabled={this.props.rowDetail.status === 'Idle' || this.props.rowDetail.status === 'Training' || this.props.rowDetail.status === 'None'}
              className={'pt-button pt-intent-primary'}
            >
              <WrapperFormattedMessage id="enter" />
            </button>
          </div>
        </div>
      </div>
    )
  }
}
export default injectIntl(connect(
  state => ({
    token: state.auth.token,
    pageStatus: state.modelManagement.pageStatus,
    modelDetail: state.modelManagement.modelDetail,
  }),
)(ModelDetail))
