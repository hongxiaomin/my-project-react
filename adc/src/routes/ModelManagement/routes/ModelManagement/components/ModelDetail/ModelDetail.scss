.container {
  margin-bottom: 30px;
  .slider {
  	width: 500px;
  }
  .result {
    background-color: #f5f5f5;
    width: 100%;
    height: 60px ;
    font-size: 18px;
    font-weight: bold;
    margin: 20px 0px;
    padding-top: 18px
  }
  .top-left{
    padding: 10px 30px;
    width: 100%;
    font-size: 18px;
    font-weight: bold;
    background-color: #f7f7f7;
    border-bottom: 1px solid #d7d7d7;
    border: 1px solid #d7d7d7;
    border-radius: 4px;
    margin-top: 30px;

  }
}
.button {
  height: 30px;
  margin-top: 15px;
}
.confidence {
  border: 1px solid #ccc; margin-bottom: 30px; border-top: 0px; padding-top: 20px
}
.divContainer {
  // height: 300px;
  border: 1px solid #d7d7d7;
  border-top: none;
  display: flex;
  justify-content: space-around;
}
.singleDiv{
  position: relative;
  width: 33%;
  border-right: 1px solid #d7d7d7;
  padding: 30px 50px;
}
.title{
  font-size: 18px;
  font-weight: bold;
}
.detail {
  padding-bottom: 10px;
}
.textField {
  padding: 0 20px;
  padding-top: 20px;
  display: flex;
  // justify-content: center;
  // align-items: flex-start;
}
  .key {
    color: #909090;
    width: 30%;
    // padding-right: 20px;
    // text-align: right;
  }

  .valueNoWrap {
    font-weight: bold;
    color: 333;
    width: 56%;
    overflow: hidden;
    word-break: break-all;
    white-space: nowrap;
    text-overflow: ellipsis;
  }
  .value {
    font-weight: bold;
    color: 333;
    width: 60%;
    word-break: break-all;
  }
