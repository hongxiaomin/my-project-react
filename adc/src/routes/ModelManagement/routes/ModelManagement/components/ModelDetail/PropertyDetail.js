import React from 'react'
import { connect } from 'react-redux'
import { Modal, Button } from 'antd'
import { injectIntl } from 'react-intl'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import classes from './ModelDetail.scss'

type Props = {
  isDetail: Object,
  data: Object,
  intl: Object,
}
export class TextField {
  constructor(key, value) {
    this.key = key
    this.value = value
  }

  getRender(btn) {
    return (
      <div className={classes.textField}>
        <div className={classes.key}>
          {this.key}
        </div>
        <div className={classes.value}>
          {this.value}
          {btn}
        </div>
      </div>
    )
  }
}
export class PropertyDetail extends React.Component {
  constructor() {
    super()
    this.state = {
      visible: false,
    }
  }

  componentDidMount() {
    // this.props.dispatch(getModelList())
  }

  toggle = () => {
    this.setState({
      visible: !this.state.visible,
    })
  }
  handleCancel = () => {
    this.setState({
      visible: false,
    })
  }

  props: Props

  render() {
    const { intl } = this.props
    const footer = [
      <Button key="back" size="large" onClick={this.handleCancel}>
        <WrapperFormattedMessage id="cancel" />
      </Button>,
    ]
    return (
      <div>
        <button
          style={{ position: 'absolute', right: '4%', bottom: '6%' }}
          type='button'
          onClick={this.toggle}
          disabled={this.props.isDetail.status === 'Idle' || this.props.isDetail.status === 'Training'}
          className={'pt-button pt-intent-primary'}
        >
          <WrapperFormattedMessage id="modelManagement.ModelDetail.showMore" />
        </button>
        <Modal
          closable={false}
          title={intl.formatMessage({ id: 'modelManagement.ModelDetail.modelDetails' })}
          footer={footer}
          visible={this.state.visible}
        >
          <div className={classes.detail}>
            {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.createTime' })}`, this.props.data.createTime).getRender()}
            {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.pairedProcessID' })}`, this.props.data.process.join(' / ')).getRender()}
            {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.pairedProductID' })}`, this.props.data.product.join(' / ')).getRender()}
            {new TextField(`${intl.formatMessage({ id: 'modelManagement.ModelDetail.pairedDefect' })}`, this.props.data.defect.join(' / ')).getRender()}
          </div>
        </Modal>
      </div>
    )
  }
}

export default injectIntl(connect(
  state => ({
    token: state.auth.token,
  }),
)(PropertyDetail))
