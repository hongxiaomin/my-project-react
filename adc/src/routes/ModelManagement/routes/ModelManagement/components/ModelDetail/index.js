import ModelDetail from './ModelDetail'

export default ModelDetail
