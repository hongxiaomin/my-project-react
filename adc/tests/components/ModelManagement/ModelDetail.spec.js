import React from 'react'
import { shallow, mount } from 'enzyme'

import { ModelDetail } from 'routes/ModelManagement/routes/ModelManagement/components/ModelDetail/ModelDetail'

describe('(Component) ModelDetail', () => {
  it('render normally', () => {
    const rowDetail = {
      isVerified: true,
      createTime: 'Jan 30, 2018 9:14:36 PM',
      status: 'none',
      isPublished: false,
      currentState: 'none',
      id: 1,
      isDefault: false,
      isTrained: true,
      lossRate: 0.00005,
      iteration: 5000,
      name: 'models',
      path: 'sfdfsdfsf',
      version: '1.0',
    }
    const dispatch = jest.fn()
    const modelDetail = {
      modelAttributes: { product: [], process: [], createTime: '2017-10-20 15:14:16.0', defect: [] },
      testingInfo: { testingProgress: 0, testingEndTime: ', testingBeginTime: ' },
      trainingInfo: { trainingEndTime: ', trainingBeginTime: ', trainingProgress: 0 },
    }
    const wrapper = shallow(
      <ModelDetail rowDetail={rowDetail} dispatch={dispatch} modelDetail={modelDetail} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('trainingProgress === 1', () => {
    const rowDetail = {
      isVerified: true,
      createTime: 'Jan 30, 2018 9:14:36 PM',
      status: 'none',
      isPublished: false,
      currentState: 'none',
      id: 1,
      isDefault: false,
      isTrained: true,
      lossRate: 0.00005,
      iteration: 5000,
      name: 'models',
      path: 'sfdfsdfsf',
      version: '1.0',
    }
    const dispatch = jest.fn()
    const modelDetail = {
      modelAttributes: { product: [], process: [], createTime: '2017-10-20 15:14:16.0', defect: [] },
      testingInfo: { testingProgress: 0, testingEndTime: ', testingBeginTime: ' },
      trainingInfo: { trainingEndTime: ', trainingBeginTime: ', trainingProgress: 1 },
    }
    const wrapper = shallow(
      <ModelDetail rowDetail={rowDetail} dispatch={dispatch} modelDetail={modelDetail} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('testingProgress === 1', () => {
    const rowDetail = {
      isVerified: true,
      createTime: 'Jan 30, 2018 9:14:36 PM',
      status: 'none',
      isPublished: false,
      currentState: 'none',
      id: 1,
      isDefault: false,
      isTrained: true,
      lossRate: 0.00005,
      iteration: 5000,
      name: 'models',
      path: 'sfdfsdfsf',
      version: '1.0',
    }
    const dispatch = jest.fn()
    const modelDetail = {
      modelAttributes: { product: [], process: [], createTime: '2017-10-20 15:14:16.0', defect: [] },
      testingInfo: { testingProgress: 1, testingEndTime: ', testingBeginTime: ' },
      trainingInfo: { trainingEndTime: ', trainingBeginTime: ', trainingProgress: 1 },
    }
    const wrapper = shallow(
      <ModelDetail rowDetail={rowDetail} dispatch={dispatch} modelDetail={modelDetail} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('click', () => {
    const rowDetail = {
      isVerified: true,
      createTime: 'Jan 30, 2018 9:14:36 PM',
      status: 'none',
      isPublished: false,
      currentState: 'none',
      id: 1,
      isDefault: false,
      isTrained: true,
      lossRate: 0.00005,
      iteration: 5000,
      name: 'models',
      path: 'sfdfsdfsf',
      version: '1.0',
    }
    const dispatch = jest.fn()
    const modelDetail = {
      modelAttributes: { product: [], process: [], createTime: '2017-10-20 15:14:16.0', defect: [] },
      testingInfo: { testingProgress: 0, testingEndTime: ', testingBeginTime: ' },
      trainingInfo: { trainingEndTime: ', trainingBeginTime: ', trainingProgress: 0 },
    }
    const wrapper = shallow(
      <ModelDetail rowDetail={rowDetail} dispatch={dispatch} modelDetail={modelDetail} />,
    )
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
  })
})
