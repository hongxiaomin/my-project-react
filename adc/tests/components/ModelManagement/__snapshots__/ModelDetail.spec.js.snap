// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelDetail render normally 1`] = `
<div
  className="container"
>
  <div
    className="top-left"
  >
    <span>
      Model Detail Information：
    </span>
    <div>
      models
       / Traning
      <span
        style={
          Object {
            "color": "rgb(102, 204, 0)",
          }
        }
      >
         incomplete 
      </span>
      / Testing
      <span
        style={
          Object {
            "color": "rgb(102, 204, 0)",
          }
        }
      >
         incomplete 
      </span>
    </div>
  </div>
  <div
    className="divContainer"
  >
    <div
      className="singleDiv"
    >
      <div
        className="title"
      >
        Attribute
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Create Time：
          </div>
          <div
            className="valueNoWrap"
          >
            2017-10-20 15:14:16.0
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Process ID：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Product ID：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Defect：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
      </div>
      <Connect(PropertyDetail)
        data={
          Object {
            "createTime": "2017-10-20 15:14:16.0",
            "defect": Array [],
            "process": Array [],
            "product": Array [],
          }
        }
        isDetail={
          Object {
            "createTime": "Jan 30, 2018 9:14:36 PM",
            "currentState": "none",
            "id": 1,
            "isDefault": false,
            "isPublished": false,
            "isTrained": true,
            "isVerified": true,
            "iteration": 5000,
            "lossRate": 0.00005,
            "name": "models",
            "path": "sfdfsdfsf",
            "status": "none",
            "version": "1.0",
          }
        }
      />
    </div>
    <div
      className="singleDiv"
    >
      <div
        className="title"
      >
        Training
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Training at：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Estimated remaining：
          </div>
          <div
            className="value"
          >
            , trainingBeginTime: 
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Progress：
          </div>
          <div
            className="value"
          >
            NaN%
          </div>
        </div>
      </div>
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
        style={
          Object {
            "bottom": "6%",
            "position": "absolute",
            "right": "4%",
            "width": "80px",
          }
        }
        type="button"
      >
        Enter
      </button>
    </div>
    <div
      className="singleDiv"
      style={
        Object {
          "borderRight": "none",
        }
      }
    >
      <div
        className="title"
      >
        Testing
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Testing at：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Estimated remaining：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Progress：
          </div>
          <div
            className="value"
          >
            NaN%
          </div>
        </div>
      </div>
      <button
        className="pt-button pt-intent-primary"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "bottom": "6%",
            "position": "absolute",
            "right": "4%",
            "width": "80px",
          }
        }
        type="button"
      >
        Enter
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) ModelDetail testingProgress === 1 1`] = `
<div
  className="container"
>
  <div
    className="top-left"
  >
    <span>
      Model Detail Information：
    </span>
    <div>
      models
       / Traning
      <span
        style={
          Object {
            "color": "rgb(102, 204, 0)",
          }
        }
      >
         incomplete 
      </span>
      / Testing
      <span
        style={
          Object {
            "color": "rgb(102, 204, 0)",
          }
        }
      >
         incomplete 
      </span>
    </div>
  </div>
  <div
    className="divContainer"
  >
    <div
      className="singleDiv"
    >
      <div
        className="title"
      >
        Attribute
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Create Time：
          </div>
          <div
            className="valueNoWrap"
          >
            2017-10-20 15:14:16.0
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Process ID：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Product ID：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Defect：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
      </div>
      <Connect(PropertyDetail)
        data={
          Object {
            "createTime": "2017-10-20 15:14:16.0",
            "defect": Array [],
            "process": Array [],
            "product": Array [],
          }
        }
        isDetail={
          Object {
            "createTime": "Jan 30, 2018 9:14:36 PM",
            "currentState": "none",
            "id": 1,
            "isDefault": false,
            "isPublished": false,
            "isTrained": true,
            "isVerified": true,
            "iteration": 5000,
            "lossRate": 0.00005,
            "name": "models",
            "path": "sfdfsdfsf",
            "status": "none",
            "version": "1.0",
          }
        }
      />
    </div>
    <div
      className="singleDiv"
    >
      <div
        className="title"
      >
        Training
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Training at：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Done at：
          </div>
          <div
            className="value"
          >
            , trainingBeginTime: 
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Progress：
          </div>
          <div
            className="value"
          >
            NaN%
          </div>
        </div>
      </div>
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
        style={
          Object {
            "bottom": "6%",
            "position": "absolute",
            "right": "4%",
            "width": "80px",
          }
        }
        type="button"
      >
        Enter
      </button>
    </div>
    <div
      className="singleDiv"
      style={
        Object {
          "borderRight": "none",
        }
      }
    >
      <div
        className="title"
      >
        Testing
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Testing at：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Done at：
          </div>
          <div
            className="value"
          >
            , testingBeginTime: 
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Progress：
          </div>
          <div
            className="value"
          >
            NaN%
          </div>
        </div>
      </div>
      <button
        className="pt-button pt-intent-primary"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "bottom": "6%",
            "position": "absolute",
            "right": "4%",
            "width": "80px",
          }
        }
        type="button"
      >
        Enter
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) ModelDetail trainingProgress === 1 1`] = `
<div
  className="container"
>
  <div
    className="top-left"
  >
    <span>
      Model Detail Information：
    </span>
    <div>
      models
       / Traning
      <span
        style={
          Object {
            "color": "rgb(102, 204, 0)",
          }
        }
      >
         incomplete 
      </span>
      / Testing
      <span
        style={
          Object {
            "color": "rgb(102, 204, 0)",
          }
        }
      >
         incomplete 
      </span>
    </div>
  </div>
  <div
    className="divContainer"
  >
    <div
      className="singleDiv"
    >
      <div
        className="title"
      >
        Attribute
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Create Time：
          </div>
          <div
            className="valueNoWrap"
          >
            2017-10-20 15:14:16.0
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Process ID：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Product ID：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Paired Defect：
          </div>
          <div
            className="valueNoWrap"
          />
        </div>
      </div>
      <Connect(PropertyDetail)
        data={
          Object {
            "createTime": "2017-10-20 15:14:16.0",
            "defect": Array [],
            "process": Array [],
            "product": Array [],
          }
        }
        isDetail={
          Object {
            "createTime": "Jan 30, 2018 9:14:36 PM",
            "currentState": "none",
            "id": 1,
            "isDefault": false,
            "isPublished": false,
            "isTrained": true,
            "isVerified": true,
            "iteration": 5000,
            "lossRate": 0.00005,
            "name": "models",
            "path": "sfdfsdfsf",
            "status": "none",
            "version": "1.0",
          }
        }
      />
    </div>
    <div
      className="singleDiv"
    >
      <div
        className="title"
      >
        Training
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Training at：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Done at：
          </div>
          <div
            className="value"
          >
            , trainingBeginTime: 
          </div>
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Progress：
          </div>
          <div
            className="value"
          >
            NaN%
          </div>
        </div>
      </div>
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
        style={
          Object {
            "bottom": "6%",
            "position": "absolute",
            "right": "4%",
            "width": "80px",
          }
        }
        type="button"
      >
        Enter
      </button>
    </div>
    <div
      className="singleDiv"
      style={
        Object {
          "borderRight": "none",
        }
      }
    >
      <div
        className="title"
      >
        Testing
      </div>
      <div
        className="detail"
      >
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Testing at：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Estimated remaining：
          </div>
          <div
            className="value"
          />
        </div>
        <div
          className="textField"
        >
          <div
            className="key"
          >
            Progress：
          </div>
          <div
            className="value"
          >
            NaN%
          </div>
        </div>
      </div>
      <button
        className="pt-button pt-intent-primary"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "bottom": "6%",
            "position": "absolute",
            "right": "4%",
            "width": "80px",
          }
        }
        type="button"
      >
        Enter
      </button>
    </div>
  </div>
</div>
`;
